oidc/oidc.go

```diff
@@ -105,26 +105,29 @@ type Provider struct {
 	// Raw claims returned by the server.
 	rawClaims []byte
 
-	// Guards all of the following fields.
-	mu sync.Mutex
 	// HTTP client specified from the initial NewProvider request. This is used
 	// when creating the common key set.
 	client *http.Client
+
+	// Guards all of the following fields.
+	mu sync.RWMutex
```

Comment:
Using an `RWMutex` instead, because we only do reads after the initial write. This should eliminate any lock contention.